README.md

```diff
@@ -238,76 +238,119 @@ the DRAM implementation. However, here as well we will encounter are a few thing
-- We can remove all batches where `last` is smaller than the last checkpoint we need
-- We also need to make sure we don't merge any batches across commits.
-
-This is similar to the log approach, but we don't need to store the undo log as we can just use the timestamps to
-figure out which batches to remove. While we can write e.g., earliest as part of writing the batch, I'm not sure
-how/where we write the `last` timestamp (@Leonid).
-
-One of the control plane APIs will be a transaction API to ensure consistency of all DBSP Spine (and Layer) state. 
-Another API will be added to restore to a previous checkpoint.
+Persistence is exposed through a control plane API added to the circuit handle. The control plane API involves:
+
+- providing a storage location (directory)
+- an API call for taking a checkpoint of the circuit
+- an API call to delete (the oldest available) checkpoint
+- a configuration option when initializing the circuit to resume from a given storage location and checkpoint
+
+Next we will explain these operations in more detail.
+
+#### Providing a storage location
+
+The circuit will store its persistent state in a set of files and sub-directories in a user-provided path.
+The circuit will take full control of this location and add/remove files/directories over time. It will also ensure that
+the directory is locked as to prevent accidentially running two processes/circuits that use the same directory.
+
+The layout of a circuit storage location looks like this:
+
+- `checkpoints.feldera`: A file containing a list of available checkpoints (uuid, ordered by time taken), and for
+  every checkpoint an optional identifier, corresponding circuit step id, and fingerprint of the circuit.
+- `*.mut`: A partially written file, that may exist during adding a new checkpoint/writing a new batch.
+  The `.mut` will eventually be removed once the file is fully written. While it is not necessary it makes it a
+  bit safer to get rid of incomplete files when recovering from a failure.
+- `<uuid>.feldera`: Batch files used by various spines of a circuit.
+- `<uuid>/`: Directories that correspond to a checkpoint and have checkpoint specific files (note that a batch
+  may be used by many checkpoints, hence they live in the base directory).
+- `<uuid>/pspine-batches-<persistent-id>.dat`: There is one for every spine in the circuit. These are rkyv files
+  of type `Vec<String>` and they contain just a list of batch files in by the spine for this given checkpoint.
+- `<uuid>/psine-<persistent-id>.dat`: This contains a rkyv representation of all the state necessary to completely
+  reconstruct `struct Spine`.
+
+Q: Why do we have two set of files, `pspine-*.dat` and `pspine-batches-*.dat` for spine meta-data: It is easier to 
+read `pspine-batches-*.dat` from anywhere in the code as it doesn't require me to know the exact generic type of \
```

Comment:
"me"?
I think you mean that the file formats is baked into the Rust code, and it cannot be interpreted in the absence of the original program that wrote them.